packages/forms/src/components/copy-to-clipboard-button/CopyToClipboardButton.tsx

```diff
@@ -0,0 +1,31 @@
+import { faCopy } from "@fortawesome/free-regular-svg-icons/faCopy";
+import { useTimeoutState } from "@stenajs-webui/core";
+import { FlatButton, FlatButtonProps } from "@stenajs-webui/elements";
+import { Tooltip } from "@stenajs-webui/tooltip";
+import * as React from "react";
+import { useCallback } from "react";
+
+export interface CopyToClipboardButtonProps<T> {
+  value: T;
+  tooltipLabel?: string;
+  size?: FlatButtonProps["size"];
+}
+
+export function CopyToClipboardButton<T>({
+  value,
+  tooltipLabel = "Copied to clipboard!",
+  size = "small",
+}: CopyToClipboardButtonProps<T>) {
+  const [visible, setVisible] = useTimeoutState(false, 2000);
+
+  const onClick = useCallback(async () => {
+    await navigator.clipboard.writeText(String(value));
+    setVisible(true);
+  }, [setVisible, value]);
```

Comment:
```suggestion
export function CopyToClipboardButton({
  getValue,
  tooltipLabel = "Copied to clipboard!",
  size = "small",
}) {
  const [visible, setVisible] = useTimeoutState(false, 2000);

  const onClick = useCallback(async () => {
    await navigator.clipboard.writeText(getValue());
    setVisible(true);
  }, [setVisible, value]);
```

Maybe something like this instead?

Reply:
## getValue
In a backend environment, yes, always. In React, no?

## not string value
I thought about it, but I don't want `toString` to be implicit. It is better to force the user to reason about the string representation.

Reply:
I removed implicit `toString` and generic. Also the prop is optional and button is disabled if not passed.
No getter, that is nice but not in React :)